public/app/features/dashboard-scene/scene/DashboardSceneUrlSync.ts

```diff
@@ -84,7 +86,29 @@ export class DashboardSceneUrlSync implements SceneObjectUrlSyncHandler {
       update.viewPanelScene = undefined;
     }
 
+    // Handle edit panel state
+    if (typeof values.editPanel === 'string') {
+      const panel = findVizPanelByKey(this._scene, values.editPanel);
+      if (!panel) {
+        return;
+      }
+
+      this._scene.onEnterEditMode();
```

Comment:
if we are to call this function like this here we need to add a check to onEnterEditMode that checks if we are already in edit mode